packages/react-native-editor/src/index.js

```diff
@@ -69,14 +76,38 @@ const setupInitHooks = () => {
 				postType = 'post';
 			}
 
+			if ( colors === undefined ) {
+				colors = SETTINGS_DEFAULTS.colors;
+			} else {
+				const validColors = colors.filter( ( c ) => c.color );
+				if ( validColors.length === 0 ) {
+					colors = SETTINGS_DEFAULTS.colors;
+				} else if ( validColors.length < colors.length ) {
+					// Filter out invalid colors
+					colors = validColors;
+				}
+			}
+
+			if ( gradients === undefined ) {
+				gradients = SETTINGS_DEFAULTS.gradients;
+			} else {
+				const validGradients = gradients.filter( ( c ) => c.gradient );
+				if ( validGradients.length === 0 ) {
+					gradients = SETTINGS_DEFAULTS.gradients;
+				} else if ( validGradients.length < gradients.length ) {
+					// Filter out invalid gradients
+					gradients = validGradients;
+				}
+			}
+
```

Comment:
This looks good overall. Since it's the same code in both places I wonder if we would be better off creating a helper that both modules could use. Maybe a function in the block-editor component as both spots reference that already